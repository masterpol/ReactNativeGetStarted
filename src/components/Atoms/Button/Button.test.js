import React from 'react'
import { View } from 'react-native'
import { Button, ButtonOrange, ButtonRed } from './'
import Theme from '_themes'
import { fireEvent } from '@testing-library/react-native'

describe('Login Screen', () => {
  let props

  beforeEach(() => {
    props = {
      onPress: jest.fn(),
      children: <View />,
    }
  })

  describe('structure', () => {
    it('should match the snapshot for Button', () => {
      const wrapper = render(
        <Theme>
          <Button testID="button-test" {...props} />
        </Theme>,
      )
      expect(wrapper).toMatchSnapshot()
    })

    it('should match the snapshot for ButtonOrange', () => {
      const wrapper = render(
        <Theme>
          <ButtonOrange testID="buttonOrange-test" {...props} />
        </Theme>,
      )
      expect(wrapper).toMatchSnapshot()
    })

    it('should match the snapshot for ButtonRed', () => {
      const wrapper = render(
        <Theme>
          <ButtonRed testID="buttonRed-test" {...props} />
        </Theme>,
      )
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('actions', () => {
    it('should call onPress action', () => {
      const { getByTestId } = render(
        <Theme>
          <Button testID="buttonRed-test" {...props} />
        </Theme>,
      )

      fireEvent.press(getByTestId('buttonRed-test'))
      expect(props.onPress).toBeCalled()
    })
  })
})
