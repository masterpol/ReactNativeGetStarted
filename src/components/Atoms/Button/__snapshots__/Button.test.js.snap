// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button structure should match the snapshot for Button 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "#2196f3",
        "borderRadius": 0,
      },
      Object {},
    ]
  }
  testID="button-test"
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "paddingBottom": 10,
          "paddingLeft": 20,
          "paddingRight": 20,
          "paddingTop": 10,
        },
        undefined,
      ]
    }
  >
    <View />
  </View>
</View>
`;

exports[`Button structure should match the snapshot for ButtonOrange 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "#ffa726",
        "borderRadius": 0,
      },
      Array [
        Object {},
        undefined,
      ],
    ]
  }
  testID="buttonOrange-test"
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "paddingBottom": 10,
          "paddingLeft": 20,
          "paddingRight": 20,
          "paddingTop": 10,
        },
        undefined,
      ]
    }
  >
    <View />
  </View>
</View>
`;

exports[`Button structure should match the snapshot for ButtonRed 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "#f44336",
        "borderRadius": 0,
      },
      Array [
        Object {},
        undefined,
      ],
    ]
  }
  testID="buttonRed-test"
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "paddingBottom": 10,
          "paddingLeft": 20,
          "paddingRight": 20,
          "paddingTop": 10,
        },
        undefined,
      ]
    }
  >
    <View />
  </View>
</View>
`;
